### DISCUSSION TITLE
Is there a way to disconnect a consumer based on it's backlog size?

### DISCUSSION THREAD

**Discussion by ghost** [apache/pulsar#22265] [Q&A]
Suppose for some reason a consumer stopped processing the messages but didn't close its connection and its backlog is queueing up.

Is there a way for a broker to disconnect the lagging consumer based on the consumer's backlog size?

Or maybe if a threshold is reached, can the broker stop sending message to that specific consumer?

I thought the "receiver queue size" configuration for the consumer handled the latter scenario, but it seems like the configuration is not what I was looking for.

**Comment by sherlock-lin** [comment]
One question I have is whether disconnecting consumers if the messages are already backlogged will cause the backlog to increase

**Comment by dao-jun** [comment]
From my understanding, NO

**Comment by lhotari** [comment]
> Suppose for some reason a consumer stopped processing the messages but didn't close its connection and its backlog is queueing up.
> 
> Is there a way for a broker to disconnect the lagging consumer based on the consumer's backlog size?
> 
> Or maybe if a threshold is reached, can the broker stop sending message to that specific consumer?

@kunwyoo Thanks for sharing. This could be a useful feature. Is this pattern something that you have used with other messaging systems? Any references from good designs in this area?

**Reply by ghost** [reply]
Nope. It's just a feature that I hope to be included.
